### PR TITLE
Create why-webpack

### PR DESCRIPTION
**Pull request by debs-obrien**
This is just a suggestion and needs tweaking, suggestions, additions etc, Just let me know what you guys think and what can be added to it, taken away etc etc. All suggestions extremely welcome. I think this page is necessary and will help a lot but I could be missing something or could have got something wrong as I went through Seans workshops taking notes of what he said. Anyway hope it is at least some what useful

### CODE REVIEW

File: src/content/concepts/why-webpack [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+title: Why Webpack
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: Please use lowercase `webpack`

File: src/content/concepts/why-webpack [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+sort: 7
+contributors:
+- debs_obrien
+---
+
+##Why webpack?
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: Please leave a space between `#` and the title: `# Why webpack?`

File: src/content/concepts/why-webpack [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+##IIFE's - Immediately invoked function expressions
+
+IIFE's solve big problems with regards to scoping. Every file is treated as an IIFE which allows us to concatenate or safely combine files without concern of scope collision. Wow, we can now write 100 files and squish them together without concerns that we are going to knock out a bunch of variables and scopes.
+
+This lead to tools like Make, Gulp, Grunt, Broccoli, Brunch and StealJS being created. The purpose of these tools/task runners is to concatenate the files together in order to solve these problems. However there are still problems. With these tools a full rebuild is needed every time. Anytime you want to change one file you have to rebuild the whole thing. This also leads to a lot of dead code. Concat doesn't help the usages across files. If you are concatenating files together how do you remove code your not using? How do you even know what code your not using?
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `Concat doesn't help the usages across files`, it would be good to rephrase this oration, maybe changing Concat to Concatenate.

Also, please remember this documentation is used by thousands non native, non advanced English speakers, it really helps use simple words.

Comment (montogeek) [reply src/content/concepts/why-webpack]: `How do you even know what code your not using?` -> `How do you even know what code you are not using?`

File: src/content/concepts/why-webpack [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+When your in scripts there is no way. If your only using one function from lodash or one date utility from moment.js you are literally adding the entire library and just squishing it together. Lot's of IIFE's are slow. Also how do you lazy load with a task runner? You don't. There is no way. Whatever your building your shipping all the code down the pipeline even if your app maybe only uses 10% of it to load and get your initial experience. As you can see there are a lot of problems with just shoving scripts together.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: Not really sure what do you want to express with `When your in scripts there is no way.`.

Comment (montogeek) [reply src/content/concepts/why-webpack]: `If your only` -> `In you are only`

Comment (montogeek) [reply src/content/concepts/why-webpack]: What do you think changing `you are literally` -> `you are actually`, it express better your intention.

Comment (montogeek) [reply src/content/concepts/why-webpack]: `Whatever your building your shipping` -> `Whatever you are building you are shipping`

File: src/content/concepts/why-webpack [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+
+##Birth of JavaScript Modules happened thanks to node.js
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `Node.js` is the correct name

File: src/content/concepts/why-webpack [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+If your using webpack your using node as webpack relies on node. Node is JavaScript that runs on the server. So how do you load JavaScript if there is no DOM? How do you add a script tag if there is no html in node?
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `If your using` -> `If you are using`

File: src/content/concepts/why-webpack [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+CommonJS came about and introduced require which allows you to inject other pieces of a module into the current module. This solves scope issues without having to to use IIFE's. This helps us to see exactly what modules are used.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: Please inclose require -> `require`

Comment (montogeek) [reply src/content/concepts/why-webpack]: Repeated word: `to to`

File: src/content/concepts/why-webpack [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+
+##Npm + node + modules -- mass distribution
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `npm` is the correct name

File: src/content/concepts/why-webpack [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+JavaScript is taking over the world as a language, as a platform and as a way to rapidly develop and create fast running applications. 
+
+But there is no browser support for commonJS. There are no live bindings. There are problems with circular references. Sync module resolution loader is slow. Bundlers/Linkers such as Browserify, RequireJS, SystemJS came about to solve these issues and allow you to write CommonJS modules which get bundled so that it works in the web.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `CommonJS` is the correct name

Comment (montogeek) [reply src/content/concepts/why-webpack]: Could you please find a link regarding `live bindings`, not everybody knows about ES modules spec details.
Maybe a blog post that explain them.

Comment (montogeek) [reply src/content/concepts/why-webpack]: What do you think about rephrasing ` get bundled so that it works in the web.` -> `get bundled, making it possible to run them in the Web`?

File: src/content/concepts/why-webpack [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+But with CommonJS there is no lazy loading. All bundles are loaded upfront. CommonJS is too dynamic for lazy loading, too much bloat, it has awkward non standard syntax and has no real module system.
+
+
+##ESM - Ecma Script Modules
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `ECMAScript` is the correct name.

File: src/content/concepts/why-webpack [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+ESM provides reusability, encapsulation, is organized and convenient but there is no ESM for node. How do they work in the browser? Incredibly slow, therefore not an option.
+
+Every library is different, library authors use the module types that they like and choose and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `like and choose and this is` sounds really weird. What do you think about: `library authors choose a module type they like`?

File: src/content/concepts/why-webpack [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+Every library is different, library authors use the module types that they like and choose and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it. 
+
+
+##Wouldn't it be nice….
+
+
+We now need something that not only lets us write modules but we need something that lets us support any module format at least until we can get to ESM, and that can handle resources and other assets at the same time.
+
+This is why webpack was born. webpack is a module bundler. It lets you write any module format or a mixed format and compiles them for the browser. It supports static async bundling (code splitting) where you can create seperate lazy loaded bundles at build time and extra optimization with it's rich, vast ecosystem. Today it is the most performant way to ship JavaScript and has 14 million downloads a month.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: Typo `separate`

Comment (montogeek) [reply src/content/concepts/why-webpack]: `with it's rich,` -> `with its rich,`

File: src/content/concepts/why-webpack [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+sort: 7
+contributors:
+- debs_obrien
+---
+
+##Why webpack?
+
+To understand why you should use webpack you first have to understand the origins of how we use JavaScript on the web. 
+
+JavaScript is just a script and there are 2 ways to use it in the browser, loaded as a script tag or written directly inside your html. The problems that this causes is that it doesn't scale. You either have too many scripts that you try to load from your script tags in the html and this causes a huge performance problem. There is only a certain amount of concurrent requests that can fetch data at a single time so breaking your application into a hundred JavaScript files and loading them in the browser is a huge performance problem. One big JS file results in unmaintainable scripts that causes problems in scope, size, readability, fragility and monolith files.
```
Comment (montogeek) [inline src/content/concepts/why-webpack]: `The problems that this causes is that it doesn't scale.` -> `, this is a problem, it doesn't scale.`

Comment (montogeek) [reply src/content/concepts/why-webpack]: Please uppercase `html`: `HTML`.

File: src/content/concepts/why-webpack (hunk 2 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 13 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 3 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 4 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 6 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 7 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 9 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: please check link to see if its ok

File: src/content/concepts/why-webpack (hunk 10 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 11 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 12 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 8 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 1 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

**Comment by debs-obrien** [comment]
also I added sort: 7 can you tell me if this is where it should go or if it should have another sort number

File: src/content/concepts/why-webpack (hunk 9 above)
Comment (montogeek) [reply src/content/concepts/why-webpack]: @debs-obrien It is, thank you

**Comment by jeremenichelli** [comment]
This looks great Debbie. I'll give it a read tomorrow with more time and come back with comments. Thanks for putting this through.

File: src/content/concepts/why-webpack [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+If you are only using one function from lodash or one date utility from moment.js you are actually adding the entire library and just squishing it together. Lot's of IIFE's are slow. Also how do you lazy load with a task runner? You don't. There is no way. Whatever you are building you are shipping all the code down the pipeline even if your app maybe only uses 10% of it to load and get your initial experience. As you can see there are a lot of problems with just shoving scripts together.
+
+
+## Birth of JavaScript Modules happened thanks to Node.js
+
+If your using webpack your using Node.js as webpack relies on Node.js. Node.js is JavaScript that runs on the server. So how do you load JavaScript if there is no DOM? How do you add a script tag if there is no html in Node.js?
+
+CommonJS came about and introduced 'require' which allows you to inject other pieces of a module into the current module. This solves scope issues without having to use IIFE's. This helps us to see exactly what modules are used.
```
Comment (EugeneHlushko) [inline src/content/concepts/why-webpack]: i suggest putting it into \` vs single quotes to get `require`

Also we could improve the readability by making it:
```diff
to use IIFE's
- . This helps us to see
+ and helps us to see exactly which modules are used.
```

File: src/content/concepts/why-webpack [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## npm + Node.js + modules -- mass distribution
+
+JavaScript is taking over the world as a language, as a platform and as a way to rapidly develop and create fast running applications. 
+
+But there is no browser support for CommonJS. There are no live bindings(https://medium.com/webpack/the-state-of-javascript-modules-4636d1774358). There are problems with circular references. Sync module resolution loader is slow. Bundlers/Linkers such as Browserify, RequireJS, SystemJS came about to solve these issues and allow you to write CommonJS modules which get bundled, making it possible to run them in the Web.
```
Comment (EugeneHlushko) [inline src/content/concepts/why-webpack]:
```
[live bindings](https://medium.com/webpack/the-state-of-javascript-modules-4636d1774358)
```
Will make `live bindings` a link

File: src/content/concepts/why-webpack [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+But with CommonJS there is no lazy loading. All bundles are loaded upfront. CommonJS is too dynamic for lazy loading, too much bloat, it has awkward non standard syntax and has no real module system.
```
Comment (EugeneHlushko) [inline src/content/concepts/why-webpack]: This can get criticized, i wouldnt be that aggressive to say `too dynamic, too much bloat` without bringing the arguments in. But it is up to the author i guess so not requesting a change here, just my thoughts.

File: src/content/concepts/why-webpack [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+ESM provides reusability, encapsulation, is organized and convenient but there is no ESM for Node.js. How do they work in the browser? Incredibly slow, therefore not an option.
+
+Every library is different, library authors choose a module type they like and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it. 
+
+
+## Wouldn't it be nice….
+
+
+We now need something that not only lets us write modules but we need something that lets us support any module format at least until we can get to ESM, and that can handle resources and other assets at the same time.
```
Comment (EugeneHlushko) [inline src/content/concepts/why-webpack]: This paragraph... When i read it over and over i cannot understand whats wrong... Maybe go with:
Wouldnt it be nice...
To have something that will not only let us write modules but also support any module format (at least until we get to ESM) and that can handle resources and assets at the same time.

File: src/content/concepts/why-webpack (hunk 16 above)
Comment (dhruvdutt) [reply src/content/concepts/why-webpack]: I agree with @EugeneHlushko. Keep it light and highlight the shortcomings of CommonJS. "bloat" is very aggressive. 😅

File: src/content/concepts/why-webpack [hunk 18]
```diff
@@ -0,0 +1,55 @@
+---
+title: Why webpack
+sort: 7
+contributors:
+- debs_obrien
+---
+
+## Why webpack?
+
+To understand why you should use webpack you first have to understand the origins of how we use JavaScript on the web.
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 16 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: ok, how is this:
CommonJS doesn't allow for lazy loading, it has non standard syntax and has no real module system.

File: src/content/concepts/why-webpack (hunk 17 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: ok sounds good

**Comment by EugeneHlushko** [comment]
Yikes :D great spot @jeremenichelli

File: src/content/concepts/why-webpack.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+## ESM - ECMAScript Modules
+
+ESM provides reusability, encapsulation, is organized and convenient but there is no ESM for Node.js. How do they work in the browser? Incredibly slow, therefore not an option.
+
+Every library is different, library authors choose a module type they like and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it. 
+
+
+## Wouldn't it be nice….
+
+
```
Comment (montogeek) [inline src/content/concepts/why-webpack.md]: Please delete this blank line.

File: src/content/concepts/why-webpack [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+sort: 7
+contributors:
+- debs-obrien
+---
+
+## Why webpack?
+
+To understand why you should use webpack lets do a recap of how we use JavaScript on the web. The origins of how we use JavaScript on the web. 
+
+JavaScript is just a script and there are 2 ways to use it in the browser, loaded as a script tag or written directly inside your html. The problems that this causes is that it doesn't scale. You either have too many scripts that you try to load from your script tags in the html and this causes a huge performance problem. There is only a certain amount of concurrent requests that can fetch data at a single time so breaking your application into a hundred JavaScript files and loading them in the browser is a huge performance problem. One big JS file results in unmaintainable scripts that causes problems in scope, size, readability, fragility and monolith files.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack]: I think the beginning is a little off, I would go with something more than:

> There are two ways to run JavaScript in a browser. First, include a script for each functionality you want to implement, the _issue_ is that the solution is hard to scale as loading too many scripts causes a network bottleneck.

Goo to mention first first, to use the word _issue_ instead of _problem_ when possible. Second, it's important to not say _performance_ when is actually a network anti-pattern, so we can differentiate other features like `ModuleConcatenation` from webpack that does actually improve performance of the script indeed.

I would start the final part with:

> The other alternative is load a big `.js` file containing all your project code, but this results in...

File: src/content/concepts/why-webpack.md (hunk 19 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack.md]: done. should the last one why webpack be a smaller title ### for example so it doesn't get the line underneath it ??

File: src/content/concepts/why-webpack [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## IIFE's - Immediately invoked function expressions
+
+IIFE's solve big problems with regards to scoping. Every file is treated as an IIFE which allows us to concatenate or safely combine files without concern of scope collision. Wow, we can now write 100 files and squish them together without concerns that we are going to knock out a bunch of variables and scopes.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack]: Again here, I would rephrase to _IIFEs solve scoping issues for large projects_.

The following should go more from the structure of the file to the benefit so users can understand better, for example:

> When script files are wrapped by an IIFE, you can safely concatenate...

File: src/content/concepts/why-webpack.md (hunk 19 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack.md]: also travis build failed, can you help me understand why it failed?

Comment (montogeek) [reply src/content/concepts/why-webpack.md]: Error is: MD013 Line length
Rule doc: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md013---line-length
Our config: https://github.com/webpack/webpack.js.org/blob/master/.markdownlint.json#L5

That line is longer than 600 characters.

Comment (EugeneHlushko) [reply src/content/concepts/why-webpack.md]: doesnt that get caught in pre-commit hook...

File: src/content/concepts/why-webpack.md [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+sort: 7
+contributors:
+- debs-obrien
+---
+
+## Why webpack?
+
+To understand why you should use webpack lets do a recap of how we use JavaScript on the web. The origins of how we use JavaScript on the web. 
+
+JavaScript is just a script and there are 2 ways to use it in the browser, loaded as a script tag or written directly inside your html. The problems that this causes is that it doesn't scale. You either have too many scripts that you try to load from your script tags in the html and this causes a huge performance problem. There is only a certain amount of concurrent requests that can fetch data at a single time so breaking your application into a hundred JavaScript files and loading them in the browser is a huge performance problem. One big JS file results in unmaintainable scripts that causes problems in scope, size, readability, fragility and monolith files.
```
Comment (montogeek) [inline src/content/concepts/why-webpack.md]: @debs-obrien We would need to break up this line into smaller lines.

File: src/content/concepts/why-webpack.md [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+JavaScript is taking over the world as a language, as a platform and as a way to rapidly develop and create fast running applications. 
+
+But there is no browser support for CommonJS. There are no [live bindings](https://medium.com/webpack/the-state-of-javascript-modules-4636d1774358). There are problems with circular references. Sync module resolution loader is slow. Bundlers/Linkers such as Browserify, RequireJS, SystemJS came about to solve these issues and allow you to write CommonJS modules which get bundled, making it possible to run them in the Web.
+
+But with CommonJS there is no lazy loading. All bundles are loaded upfront. CommonJS doesn't allow for lazy loading, it has non standard syntax and has no real module system.
+
+
+## ESM - ECMAScript Modules
+
+ESM provides reusability, encapsulation, is organized and convenient but there is no ESM for Node.js. How do they work in the browser? Incredibly slow, therefore not an option.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: This sounds like a really strong take, people have put effort in making modules happen in JavaScript. I would change it to something like:

> Good news for web projects is that modules are becoming an official feature in ECMAScript standard, though browser support is still short and early implementations show that bundling is still faster and recommended today.

File: src/content/concepts/why-webpack.md [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
+
+Every library is different, library authors choose a module type they like and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: I think this is not necessary.

File: src/content/concepts/why-webpack.md [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+
+ESM provides reusability, encapsulation, is organized and convenient but there is no ESM for Node.js. How do they work in the browser? Incredibly slow, therefore not an option.
+
+Every library is different, library authors choose a module type they like and this is just for JavaScript. Each and any other filetype until now has had to have specific ways to process it. 
+
+
+## Wouldn't it be nice….
+
+
+To have something that will not only let us write modules but also support any module format (at least until we get to ESM) and that can handle resources and assets at the same time.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: If this is continuation of the title shouldn't it begin with _... to have_

File: src/content/concepts/why-webpack.md [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+
+This is why webpack was born. webpack is a module bundler. It lets you write any module format or a mixed format and compiles them for the browser. It supports static async bundling (code splitting) where you can create separate lazy loaded bundles at build time and extra optimization with its rich, vast ecosystem. Today it is the most performant way to ship JavaScript and has 14 million downloads a month.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: Can we shorten this? I think that highlighting code splitting and explaining it over again doesn't makes sense, like:

> This is why webpack exists, it's a tool that not only let's you bundle your JavaScript applications, supporting both ESM and CommonJS, but can be extended to support all different kind of assets like images, fonts and stylesheets.

Also, claiming to be _the most performance_ at something requires proves, so I would rephrase the last paragraph.

> webpack cares a lot about performance and it's always adding and improving features like async chunk loading and prefetching to help you deliver the best possible version of your project to the user, always caring about loading times and performance.

File: src/content/concepts/why-webpack [hunk 27]
```diff
@@ -0,0 +1,55 @@
+---
+title: Why webpack
+sort: 7
+contributors:
+- debs-obrien
+---
+
+## Why webpack?
+
+To understand why you should use webpack lets do a recap of how we use JavaScript on the web. The origins of how we use JavaScript on the web.
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack (hunk 20 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: ok cool, think i put this part in right but if i deleted too much let me know

File: src/content/concepts/why-webpack (hunk 21 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack]: done

File: src/content/concepts/why-webpack.md (hunk 22 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack.md]: Ok @jeremenichelli comment helped fix that issue, but yeh it was too big

File: src/content/concepts/why-webpack.md (hunk 19 above)
Comment (debs-obrien) [reply src/content/concepts/why-webpack.md]: ok cool thanks

**Comment by debs-obrien** [comment]
should I be adding all your names in to the contributions too?

File: src/content/concepts/why-webpack.md [hunk 28]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## IIFE's - Immediately invoked function expressions
+
+IIFE's solve big problems with regards to scoping. Every file is treated as an IIFE which allows us to concatenate or safely combine files without concern of scope collision. Wow, we can now write 100 files and squish them together without concerns that we are going to knock out a bunch of variables and scopes.
+
+This lead to tools like Make, Gulp, Grunt, Broccoli, Brunch and StealJS being created. The purpose of these tools/task runners is to concatenate the files together in order to solve these problems. However there is much more to it. With these tools a full rebuild is needed every time. Anytime you want to change one file you have to rebuild the whole thing. This also leads to a lot of dead code. Concatenate doesn't help the usages across files. If you are concatenating files together how do you remove code your not using? How do you even know what code you are not using?
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply src/content/concepts/why-webpack.md]: done

**Comment by EugeneHlushko** [comment]
@debs-obrien i actually do that sometimes (add reviewers) when people contribute good suggestions during PR review, so when you feel like it - add it!

**Comment by debs-obrien** [comment]
@EugeneHlushko great, I have just added names as although I may have done most of the work I feel a lot of value was added as it wasn't just corrections but complete rewording and I don't feel I should take all the credit for that. But if anyone doesn't want to have their name there they can just ask to be removed :)

**Comment by montogeek** [comment]
@jeremenichelli Please review

File: src/content/concepts/why-webpack.md [hunk 29]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## ESM - ECMAScript Modules
+
+The good news for web projects is that modules are becoming an official feature in ECMAScript standard, though browser support is still short and early implementations show that bundling is still faster and recommended today.
+
+
+## Wouldn't it be nice…
+
+to have something that will not only let us write modules but also support any module format (at least until we get to ESM) and that can handle resources and assets at the same time.
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: Missing capitalized first letter, I know the sentence's coming from the title but user might think there's a typo or word missing. If not we can add `... to` as the beginning of the paragraph that would also work.

File: src/content/concepts/why-webpack.md [hunk 30]
```diff
@@ -0,0 +1,59 @@
...
+
+This is why webpack exists. It's a tool that not only let's you bundle your JavaScript applications, supporting both ESM and CommonJS, but can be extended to support all different kind of assets like images, fonts and stylesheets.
+
+webpack cares a lot about performance and it's always adding and improving features like async chunk loading and prefetching to help you deliver the best possible version of your project to the user, always caring about loading times and performance.
+
+
+### Why webpack? That's why!!!!
```
Comment (jeremenichelli) [inline src/content/concepts/why-webpack.md]: I would remove this, but just my opinion.

**Comment by jeremenichelli** [comment]
Left two minor comments, otherwise I think is ready for merge. We can alter iterate on it.

**Comment by debs-obrien** [comment]
@jeremenichelli fixed those 2 issues as yes think you were right. The ending why webpack? That's why is probably better in a presentation but not in the written docs and I see what you mean by the ...to

**Comment by jeremenichelli** [comment]
Cool! Thanks a lot for addressing all the comments @debs-obrien, sorry if it felt too much. Merging this now, hope to see more contributions from you 💛

**Comment by montogeek** [comment]
@jeremenichelli Please, once again, Squash and merge.
https://github.com/webpack/webpack.js.org/pull/2224#issuecomment-396212048

**Comment by jeremenichelli** [comment]
@montogeek always forget, sorry. Curious, is not squashing causing some particular issue?

**Comment by montogeek** [comment]
Yes, it is not aligned with the project standard that has been used since years ago, also, it is corrupting our Git history:
image

Which we should take care if we want to syncronize webpack and webpack docs releases in the future.

Last but not least, those commits doesn't follow https://conventionalcommits.org/, which this repository uses, that is something that I change when Squashing and merging.

**Comment by jeremenichelli** [comment]
This is too trivial to deal honestly, for both contributors and members, these _rules_ are written nowhere neither reinforce by the infrastructure of the project. We all have jobs where conventions might defer and pretending all of us will have our head in the right place will never work.

Suggestion: 
 - Force squashing in the repository Settings.
 - Apply a githook using `commitizen` or `conventional-commit-message` packages to avoid non valid messagers.
 - Add this into the README so contributors know why and how.

**Comment by montogeek** [comment]
Yep, will do

**Comment by montogeek** [comment]
Another problem is that since it is most Markdown, lot of contributors use GitHub web editor, again, we can catch those when Squashing and merging.

**Comment by debs-obrien** [comment]
@montogeek ohhh, didn't know we had a standard for commits. We use similar in work so will do it this way from now on but yeh some sort of instructions would be good as I never found that link or knew how best to write the commit message etc

no worries @jeremenichelli think it is an important page so deserves the extra work. It has been a great experience so thanks everyone for your help in getting this to go live. Appreciate it

**Comment by montogeek** [comment]
@debs-obrien No worries, sorry, it is our fault, we will enforce it.
Thanks for the contribution, it is a lot of work.
You can see it live here: https://webpack.js.org/concepts/why-webpack/ 🎆

**Comment by montogeek** [comment]
Uhmm, we oversight how our titles work, Could you please send another PR removing `## Why webpack` at the beginning?

**Comment by debs-obrien** [comment]
@montogeek I was just going to say the same :)
